### DISCUSSION TITLE
company model and nxml-mode

### DISCUSSION THREAD

**Discussion by paaguti** [company-mode/company-mode#1432] [Q&A]
Hi,
apparently, nxml mode has a capf function of its own. Is there any way to integrate that with compay mode? Alternatively, does anyone know of a company-capf for nxml mode?

Thanks

**Comment by dgutov** [comment]
> Is there any way to integrate that with compay mode?

It should just work in the default configuration. In all contexts where the capf function offers completion for `C-M-i` (default binding for `completion-at-point`), it should likewise provide completions with `company-mode` through `company-capf`.

Depending on which files you tried it with, it might not work too well outside of well-structured XML. But for HTML5, for instance, there is an option of using the corresponding RELAX NG schema, some instructions here: https://stackoverflow.com/a/1083226

Again, this applies both with and without `company-mode`.
